client/src/Components/Calculator.jsx

```diff
@@ -0,0 +1,91 @@
+import React, { useState, useEffect } from "react";
+import axios from "axios";
+import LeftPanel from "./LeftPanel";
+import GraphArea from "./GraphArea";
+import ErrorPage from "./ErrorPage";
+
+function Calculator() {
+  const [monthlyInvestment, setMonthlyInvestment] = useState(500);
+  const [investmentPeriod, setInvestmentPeriod] = useState(1);
+  const [rateOfReturn, setRateOfReturn] = useState(1);
+  const [delay, setDelay] = useState(1);
+  const [startToday, setStartToday] = useState();
+  const [delayedStart, setDelayedStart] = useState();
+  const [notionalLoss, setNotionalLoss] = useState();
+  const [err, setErr] = useState();
+
+  function changeValues(index, val) {
+    switch (index) {
+      case 0:
+        setMonthlyInvestment(val);
+        break;
+      case 1:
+        setInvestmentPeriod(val);
+        break;
+      case 2:
+        setRateOfReturn(val);
+        break;
+      case 3:
+        setDelay(val);
+        break;
+    }
+  }
+
+  //For  api
+
+  useEffect(() => {
+    async function fetchData() {
```

Comment:
need some discussion regarding this fetching function
can be done directly axios.get() , instead of making fetchData() , then calling fetchData() in use effect